haystack/utils/__init__.py

```diff
@@ -29,6 +29,6 @@
     "deserialize_callable",
     "serialize_type",
     "deserialize_type",
-    "deserialize_document_store_in_init_params_inplace",
+    "deserialize_component_in_init_params_inplace",
```

Comment:
We can create an alias above from here:

deserialize_document_store_in_init_params_inplace = deserialize_component_in_init_params_inplace

and then export both to minimize havoc this will create in downstream deps